File: SamCross/ExampleProjects/Math_Graphic/src/main/java/core/Math/Matrix/MatrixMath.java
package  core.Math.Matrix;

import org.apache.commons.math3.linear.MatrixUtils;
import org.apache.commons.math3.linear.RealMatrix;

/**
 * Created by Michal on 2018-04-08.
 */
public class MatrixMath implements IMatrixMath {
    @Override
    public IMatrix inverseMatrix(IMatrix m1) {
        return m1.createIMatrix(MatrixUtils.inverse(m1.getRealMatrix()));
    }

    @Override

    public IMatrix matrixAddition(IMatrix m1, IMatrix m2) throws InvalidDimensionException {
        if(m1.getHeight() != m2.getHeight() || m1.getWidth() != m2.getWidth()) throw new InvalidDimensionException();
        return m1.createIMatrix(m1.getRealMatrix().add(m2.getRealMatrix()));
    }

    @Override
    public IMatrix matrixMultiplication(IMatrix m1, IMatrix m2) throws InvalidDimensionException {
        if(m1.getWidth() != m2.getHeight()) throw new InvalidDimensionException();
        return m1.createIMatrix(m1.getRealMatrix().multiply(m2.getRealMatrix()));
    }

    @Override
    public IMatrix matrixSubtracting(IMatrix m1, IMatrix m2) throws InvalidDimensionException {
        if(m1.getHeight() != m2.getHeight() || m1.getWidth() != m2.getWidth()) throw new InvalidDimensionException();
        return m1.createIMatrix(m1.getRealMatrix().subtract(m2.getRealMatrix()));
    }

    @Override
    public IMatrix matrixTransposition(IMatrix m1) throws InvalidDimensionException {
        return m1.createIMatrix(m1.getRealMatrix().transpose());
    }

    @Override
    public IMatrix scalarMultiplication(IMatrix m1, double scalar) {
        return m1.createIMatrix(m1.getRealMatrix().scalarMultiply(scalar));
    }
}

File: stryker-net/ExampleProjects/Math_Graphic/src/Math_Graphic/core/math/matrix/MatrixMath.cs
using Math_Graphic.core.math.modules;

namespace Math_Graphic.core.math.matrix;

public class MatrixMath : IMatrixMath
{
    public IMatrix InverseMatrix(IMatrix m1)
    {
        return m1.CreateIMatrix(MatrixUtils.Inverse(m1.GetRealMatrix()));
    }


    public IMatrix MatrixAddition(IMatrix m1, IMatrix m2)
    {
        if (m1.GetHeight() != m2.GetHeight() || m1.GetWidth() != m2.GetWidth()) throw new InvalidDimensionException();
        return m1.CreateIMatrix(m1.GetRealMatrix().Add(m2.GetRealMatrix()));
    }

    public IMatrix MatrixMultiplication(IMatrix m1, IMatrix m2)
    {
        if (m1.GetWidth() != m2.GetHeight()) throw new InvalidDimensionException();
        return m1.CreateIMatrix(m1.GetRealMatrix().Multiply(m2.GetRealMatrix()));
    }

    public IMatrix MatrixSubtracting(IMatrix m1, IMatrix m2)
    {
        if (m1.GetHeight() != m2.GetHeight() || m1.GetWidth() != m2.GetWidth()) throw new InvalidDimensionException();
        return m1.CreateIMatrix(m1.GetRealMatrix().Subtract(m2.GetRealMatrix()));
    }

    public IMatrix MatrixTransposition(IMatrix m1)
    {
        return m1.CreateIMatrix(m1.GetRealMatrix().Transpose());
    }

    public IMatrix ScalarMultiplication(IMatrix m1, double scalar)
    {
        return m1.CreateIMatrix(m1.GetRealMatrix().ScalarMultiply(scalar));
    }
}